Content.Shared/Tools/Components/ToolRefinableComponent.cs

```diff
@@ -35,4 +37,56 @@ public sealed partial class ToolRefinableComponent : Component
     /// </summary>
     [DataField]
     public ProtoId<ToolQualityPrototype> QualityNeeded = "Welding";
+
+    /// <summary>
+    /// Sound that will be played after refine process finished.
+    /// </summary>
+    [DataField]
+    public SoundSpecifier? Sound;
+
+    /// <summary>
+    /// Refine verb text. If there is no text, verb will not be added.
+    /// </summary>
+    [DataField]
+    public LocId? VerbText;
+
+    /// <summary>
+    /// Icon for refine verb.
+    /// </summary>
+    [DataField]
+    public SpriteSpecifier? VerbIcon;
+
+    /// <summary>
+    /// Default tooltip text for refine verb.
+    /// </summary>
+    [DataField]
+    public LocId? VerbDefaultTooltip;
+
+    /// <summary>
+    /// Verb tooltip in case currently used tool
+    /// is missing required (<see cref="QualityNeeded"/>) quality.
+    /// </summary>
+    [DataField]
+    public LocId? ToolMissingQualityTooltip;
+
+    /// <summary>
+    /// Popup text to display for action performer
+    /// on refine action performed (end of DoAfter).
+    /// </summary>
+    [DataField]
+    public LocId? PopupForUser;
+
+    /// <summary>
+    /// Popup text to display for players near action performer
+    /// on action performed (end of DoAfter).
+    /// </summary>
+    [DataField]
+    public LocId? PopupForOther;
+
+    /// <summary>
+    /// Is usage of 'caution' type popup required for <see cref="PopupForUser"/> and <see cref="PopupForOther"/>,
+    /// or small popup will suffice.
+    /// </summary>
+    [DataField]
+    public bool IsUsingCautionPopup;
```

Comment:
Instead of a bool I think you could just let them decide. 
```suggestion
    public PopupType PopupType;
```